libraries/clchain/include/clchain/graphql.hpp

```diff
@@ -0,0 +1,769 @@
+      std::string result(ss.size, 0);
+      eosio::fixed_buf_stream fbs(result.data(), result.size());
+      fill_gql_schema((T*)nullptr, fbs);
+      eosio::check(fbs.pos == fbs.end, eosio::convert_stream_error(eosio::stream_error::underrun));
+      return result;
+   }
+
+   struct gql_stream
+   {
+      enum token_type
+      {
+         unstarted,
+         eof,
+         error,
+         punctuator,
+         name,
+         string,
+         integer,
+         floating,
+      };
+
+      eosio::input_stream input;
+      token_type current_type = unstarted;
+      std::string_view current_value;
+      char current_puncuator = 0;
+
+      gql_stream(eosio::input_stream input) : input{input} { skip(); }
+      gql_stream(const gql_stream&) = default;
+      gql_stream& operator=(const gql_stream&) = default;
+
+      void skip()
+      {
+         if (current_type == error)
+            return;
+         current_puncuator = 0;
+         current_value = {};
+         while (true)
+         {
+            auto begin = input.pos;
+            if (!input.remaining())
+            {
+               current_type = eof;
+               return;
+            }
+            switch (input.pos[0])
+            {
+               case '\n':
+               case '\r':
+               case ' ':
+               case '\t':
+               case ',':
+                  ++input.pos;
+                  continue;
+               case '#':
+                  while (input.remaining() && (input.pos[0] != '\n' && input.pos[0] != '\r'))
+                     ++input.pos;
+                  continue;
+               case '$':
+               case '!':
+               case '&':
+               case '(':
+               case ')':
+               case ':':
+               case '=':
+               case '@':
+               case '[':
+               case ']':
+               case '{':
+               case '|':
+               case '}':
+                  current_puncuator = input.pos[0];
+                  ++input.pos;
+                  current_value = {begin, size_t(input.pos - begin)};
+                  current_type = punctuator;
+                  return;
+               case '.':
+                  if (input.remaining() >= 3 && input.pos[1] == '.' && input.pos[2] == '.')
+                  {
+                     current_puncuator = '.';
+                     input.pos += 3;
+                     current_value = {begin, size_t(input.pos - begin)};
+                     current_type = punctuator;
+                     return;
+                  }
+                  break;
+               case '"':
+                  ++input.pos;
+                  while (input.remaining() && input.pos[0] != '"')
+                  {
+                     auto ch = *input.pos++;
+                     if (ch == '\\')
+                     {
+                        if (!input.remaining())
+                           return;
+                        ++input.pos;
```

Comment:
This tokenizer intentionally doesn't translate escapes since the majority use cases for string arguments (cursors and numerics) don't need it. If we ever have query args in the future that do need it, we can add a manually-invoked conversion.

Reply:
"\u0030" should be considered equivalent to "0".

Reply:
Added note